op-node/rollup/derive/attributes.go

```diff
@@ -101,44 +98,29 @@ func (ba *FetchingAttributesBuilder) PreparePayloadAttributes(ctx context.Contex
 			l2Parent, nextL2Time, eth.ToBlockID(l1Info), l1Info.Time()))
 	}
 
+	var upgradeTxs []hexutil.Bytes
+	if ba.cfg.IsEcotoneUpgradeDepositBlock(l2Parent.Time, nextL2Time) {
+		upgradeTxs, err = EcotoneNetworkUpgradeTransactions()
+		if err != nil {
+			return nil, NewCriticalError(fmt.Errorf("failed to build ecotone network upgrade txs: %w", err))
+		}
+	}
+
```

Comment:
Cool, I was over in `Dencun` branch wondering what the cleanest way to do this would be: it's this way, I think 👍